### PR TITLE
[CONSUL-176] Support replacing CRLF by LF on scenarios scripts

### PR DESCRIPTION
**Pull request by iberlot**
### Description
When running the tests, it must be verified that the bash and sh files are in the correct format. To do this, the directories are traversed and the line endings of those files are checked. If they are CRLF they are replaced by LF.

### PR Checklist

* [ ] find the files with crlf
* [ ] update files with crlf
* [ ] does not modify LF files
* [ ] do not modify files other than sh or bash

### CODE REVIEW

File: test/integration/connect/envoy/main_test.go [hunk 1]
```diff
@@ -22,6 +23,10 @@ var (
 func TestEnvoy(t *testing.T) {
 	flag.Parse()
 
+	if *flagWin == true {
+		travel_dir("../../../")
```
Comment (cocolavayen) [inline test/integration/connect/envoy/main_test.go]: I feel that this argument is a bit strange. Perhaps separating it into a variable might be clearer.

File: test/integration/connect/envoy/main_test.go [hunk 2]
```diff
@@ -102,3 +107,57 @@ func discoverCases() ([]string, error) {
 	sort.Strings(out)
 	return out, nil
 }
+
+
+// crlf convert functions
+
+func travel_dir(path string ){
```
Comment (cocolavayen) [inline test/integration/connect/envoy/main_test.go]: Delete empty space before the parenthesis
Add empty space after parenthesis

`func travel_dir(path string) {`

File: test/integration/connect/envoy/main_test.go [hunk 3]
```diff
@@ -102,3 +107,57 @@ func discoverCases() ([]string, error) {
...
+    files, err := ioutil.ReadDir(path)
+    if err != nil {
+        log.Fatal(err)
+    }
+    for _, fil := range files {
+        
+        v := strings.Split(fil.Name(), ".")  
+        // names := v[0]
```
Comment (cocolavayen) [inline test/integration/connect/envoy/main_test.go]: Lets' remove comments/unused lines

File: test/integration/connect/envoy/main_test.go [hunk 4]
```diff
@@ -102,3 +107,57 @@ func discoverCases() ([]string, error) {
...
+            }
+        // }
+
+        if exts == "sh" || exts == "bash" {
+            crlf_file_check(file_path)
+        }
+    }
+}
+
+func crlf_file_check(file_name string ){
```
Comment (cocolavayen) [inline test/integration/connect/envoy/main_test.go]: Same here as in the previous comment

File: test/integration/connect/envoy/main_test.go [hunk 5]
```diff
@@ -102,3 +107,57 @@ func discoverCases() ([]string, error) {
...
+    if edit := crlf_verify(text); edit != -1 {
+        crlf_normalize(file_name, text)
+    }
+    
+    if err != nil {
+        log.Fatal(err)
+    }
+}
+
+func crlf_verify(text string ) int{
```
Comment (cocolavayen) [inline test/integration/connect/envoy/main_test.go]: Same here as in the previous comment.

File: test/integration/connect/envoy/main_test.go [hunk 6]
```diff
@@ -102,3 +107,57 @@ func discoverCases() ([]string, error) {
...
+    position := strings.Index(text, "\r\n"); 
+    return position
+}
+
+func crlf_normalize(filename, text string ) {
+    text = strings.Replace(text,"\r\n","\n",-1)
+    data := []byte(text)
+
+    ioutil.WriteFile(filename, data, 0644)
+}
```
Comment (cocolavayen) [inline test/integration/connect/envoy/main_test.go]: missing empty line

File: test/integration/connect/envoy/main_test.go [hunk 7]
```diff
@@ -102,3 +107,57 @@ func discoverCases() ([]string, error) {
...
+func travel_dir(path string ){
+    files, err := ioutil.ReadDir(path)
+    if err != nil {
+        log.Fatal(err)
+    }
+    for _, fil := range files {
+        
+        v := strings.Split(fil.Name(), ".")  
+        // names := v[0]
+        exts := v[len(v)-1]
```
Comment (cocolavayen) [inline test/integration/connect/envoy/main_test.go]: a variable name like `file_extension` could be more understandable

File: test/integration/connect/envoy/main_test.go (hunk 2 above)
Comment (cocolavayen) [inline test/integration/connect/envoy/main_test.go]: I think that a function name like `check_dir_files` might be more intuitive.

File: test/integration/connect/envoy/main_test.go [hunk 8]
```diff
@@ -102,3 +107,57 @@ func discoverCases() ([]string, error) {
 	sort.Strings(out)
 	return out, nil
 }
+
+
+// crlf convert functions
+
```
Comment (cocolavayen) [inline test/integration/connect/envoy/main_test.go]: remove empty line

File: test/integration/connect/envoy/main_test.go [hunk 9]
```diff
@@ -102,3 +107,57 @@ func discoverCases() ([]string, error) {
 	sort.Strings(out)
 	return out, nil
 }
+
+
+// crlf convert functions
```
Comment (cocolavayen) [inline test/integration/connect/envoy/main_test.go]: Let's add some comments on the purpose of the functions to provide some further guidance

File: test/integration/connect/envoy/main_test.go [hunk 10]
```diff
@@ -102,3 +107,57 @@ func discoverCases() ([]string, error) {
...
+func crlf_verify(text string ) int{
+    position := strings.Index(text, "\r\n"); 
+    return position
+}
+
+func crlf_normalize(filename, text string ) {
+    text = strings.Replace(text,"\r\n","\n",-1)
+    data := []byte(text)
+
+    ioutil.WriteFile(filename, data, 0644)
```
Comment (cocolavayen) [inline test/integration/connect/envoy/main_test.go:164]: What type of permission is this?

Comment (iberlot) [reply test/integration/connect/envoy/main_test.go:164]: Owner: 6 = reading/writing
Group: 4 = reading
Rest: 4 = reading

**Review by ezfepo — APPROVED** [review]

**Pull request merged**